src/main/lrsql/system/lrs.clj

```diff
@@ -261,6 +261,12 @@
           input (admin-input/delete-admin-input account-id)]
       (jdbc/with-transaction [tx conn]
         (admin-cmd/delete-admin! backend tx input))))
+  (-ensure-account-oidc
+    [this username oidc-issuer]
+    (let [conn  (lrs-conn this)]
+      (jdbc/with-transaction [tx conn]
+        (admin-cmd/ensure-admin-oidc! backend tx {:username    username
+                                                  :oidc-issuer oidc-issuer}))))
```

Comment:
Now that I have the bigger picture, I think you should call an `ensure-admin-oidc-input` function here, then generate the primary key if need be at the op level.

Reply:
Yep, that sounds solid for ensure. I think I see what you're saying about `insert-admin!` too, because it fires that query first.

Reply:
Done for ensure on https://github.com/yetanalytics/lrsql/pull/207/commits/1e2526a4fe40d895f9a925606b31d48075510573, added the `add-primary-key` fn to util for the purpose. I'll go ahead and try a similar thing for the vanilla

Reply:
for the vanilla admin https://github.com/yetanalytics/lrsql/pull/207/commits/785148c4818620f5c3e6fd817b4040405cec4c49

Reply:
ended up reverting the changes to vanilla admin https://github.com/yetanalytics/lrsql/commit/785148c4818620f5c3e6fd817b4040405cec4c49, ran into weird spec problems that are out of scope for this anyhow. Honestly the chance of generating an unneeded ID is much lower with that than when ensuring OIDC admins.

Reply:
To quote someone wise, "well we tried."